File: palendromeNumber/palindrome.c
#include <stdio.h>
#include <stdbool.h>

int reverse(int x)
{
	long long int reversed = 0; // for big value long long required
	int last_digit;
	while (x != 0)
	{
		last_digit = x % 10;
		reversed = (reversed * 10) + last_digit;
		x = x / 10;
	}
	return reversed;
}

bool isPalindrome(int x)
{
	if (x >= 0 &&  x == reverse(x))
		return true;
	return false;
}

int main()
{
	printf("%i\n", isPalindrome(1221));
	return 0;
}

File: palendromeNumber/palindrome.py
def reverse_int(x):
	reversed_num = 0
	lastdigit = 0
	while (x != 0):
		lastdigit = x % 10
		reversed_num = (reversed_num * 10) + lastdigit
		x = x // 10 # / return float but // returns int
	return reversed_num

def isPalindrome(x):
	if (x >= 0 and x == reverse_int(x)):
		return True
	return False

print(isPalindrome(1221))